sklearn/model_selection/tests/test_validation.py

```diff
@@ -1282,6 +1321,33 @@ def test_validation_curve_cv_splits_consistency():
     assert_array_almost_equal(np.array(scores3), np.array(scores1))
 
 
+def test_validation_curve_fit_params():
+    est = SVC(random_state=0)
+    n_samples = 50
+    X, y = make_blobs(n_samples=n_samples, n_features=2, centers=10,
+                      random_state=0)
+    # sample_weight is a list
+    w = [i for i in range(n_samples)]
+    gamma_range = np.logspace(-6, -1, 5)
+    l_with_fit_params = validation_curve(est, X, y, "gamma", gamma_range,
```

Comment:
Now you need to pass everything but `est, X and y` by name: use `validation_curve(est, X, y, param_name="gamma", param_range=gamma_range, fit_params={'sample_weight': w})`